driver/main.c

```diff
@@ -634,26 +622,49 @@ static int compat_set_tracepoint(void *func, const char *probename, struct trace
 	return ret;
 }
 
-static int force_tp_set(u32 new_tp_set, u32 max_val)
+static int force_tp_set(struct ppm_consumer_t *consumer, u32 new_tp_set)
 {
 	u32 idx;
 	u32 new_val;
 	u32 curr_val;
-	int ret = 0;
-
-	if (new_tp_set == g_tracepoints_attached)
-	{
-		// ok already equal
-		return ret;
-	}
+	int cpu;
+	int ret;
```

Comment:
I would initialize all these variables just to be future proof

Reply:
We can't mix declaration and code in c89/90 :) I mean, we actually can but then any added variable must be initialized.
We try to avoid doing that in kmod!